spec/requests/retrieve_stats_spec.rb

```diff
@@ -0,0 +1,108 @@
+require "rails_helper"
+
+RSpec.describe "Retrieving stats", type: :request do
+
+  let(:account) { Account.create!(name: "Test Account") }
+
+  before do
+    account.raw_test_runs.create! payload: [
+      { name: "Test 1", status: "passed", run_time: 1.1 },
+      { name: "Test 2", status: "passed", run_time: 1.5 },
+      { name: "Test 3", status: "passed", run_time: 2.5 },
+      { name: "Test 4", status: "failed", run_time: 0.5 }
+    ]
+    account.raw_test_runs.create! payload: [
+      { name: "Test 1", status: "passed", run_time: 1.1 },
+      { name: "Test 2", status: "passed", run_time: 1.5 },
+      { name: "Test 3", status: "failed", run_time: 0.5 },
+      { name: "Test 4", status: "passed", run_time: 1.5 }
+    ]
+  end
+
+  let(:json_body) { JSON.parse(response.body) }
+
+  describe "Fetching the most frequently failing tests" do
+
+    describe "with a valid account" do
+
+      it "returns an array sorted by failure count" do
+        get "/api/v1/stats/most_failing",
+          {},
+          { "ACCEPT" => "application/json",
```

Comment:
Align the parameters of a method call if they span more than one line.<br>Redundant curly braces around a hash parameter.